### PR TITLE
Add CBMC proof for ICMP functions

### PR DESCRIPTION
**Pull request by tony-josi-aws**
Description
-----------
This PR adds CBMC proofs for the following ICMP functions:

- ProcessICMPPacket
- prvProcessICMPEchoReply
- prvProcessICMPEchoRequest

Test Steps
-----------
Ran CBMC on the proofs

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/cbmc/proofs/ICMP/ProcessICMPEchoRequest/ProcessICMPEchoRequest_harness.c [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+eFrameProcessingResult_t __CPROVER_file_local_FreeRTOS_ICMP_c_prvProcessICMPEchoRequest( ICMPPacket_t * const pxICMPPacket,
+                                                                                         const NetworkBufferDescriptor_t * const pxNetworkBuffer );
+
+void harness()
+{
+    NetworkBufferDescriptor_t xNetworkBuffer;
+
+    ICMPPacket_t * const pxICMPPacket = malloc( sizeof( ICMPPacket_t ) );
```
Comment (ActoryOu) [inline test/cbmc/proofs/ICMP/ProcessICMPEchoRequest/ProcessICMPEchoRequest_harness.c:73]: Please use safeMalloc instead.

File: test/cbmc/proofs/ICMP/ProcessICMPPacket/ProcessICMPPacket_harness.c [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_ICMP.h"
+#include "FreeRTOS_IP_Private.h"
+
```
Comment (ActoryOu) [inline test/cbmc/proofs/ICMP/ProcessICMPPacket/ProcessICMPPacket_harness.c:39]:
```suggestion

/* CBMC includes. */
```

File: test/cbmc/proofs/ICMP/ProcessICMPPacket/ProcessICMPPacket_harness.c [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+#include "cbmc.h"
+
+/* prvProcessICMPEchoRequest() is proved separately */
+eFrameProcessingResult_t __CPROVER_file_local_FreeRTOS_ICMP_c_prvProcessICMPEchoRequest( ICMPPacket_t * const pxICMPPacket,
+                                                                                         const NetworkBufferDescriptor_t * const pxNetworkBuffer )
+{
+    eFrameProcessingResult_t retVal;
```
Comment (ActoryOu) [inline test/cbmc/proofs/ICMP/ProcessICMPPacket/ProcessICMPPacket_harness.c:46]: Please FreeRTOS follow naming rule.

File: test/cbmc/proofs/ICMP/ProcessICMPPacket/ProcessICMPPacket_harness.c [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    return retVal;
+}
+
+void harness()
+{
+    NetworkBufferDescriptor_t xNetworkBuffer;
+
+    /* Allocate a ICMP packet */
+    ICMPPacket_t * const pxICMPPacket = malloc( sizeof( ICMPPacket_t ) );
```
Comment (ActoryOu) [inline test/cbmc/proofs/ICMP/ProcessICMPPacket/ProcessICMPPacket_harness.c:61]: Please use safeMalloc instead.

**Review by ActoryOu — CHANGES_REQUESTED** [review]

**Review by ActoryOu — APPROVED** [review]

**Review by shubnil — APPROVED** [review]

**Pull request merged**